### PR TITLE
Run Code formatter on Kernel.CLI

### PR DESCRIPTION
**Pull request by jjcarstens**
I should probably question the fact that the most ideal task I wanted to do this evening was format code...but this was very fulfilling!

Comments in line for some specifics I changed after the formatter that could use 👀

### CODE REVIEW

File: lib/elixir/lib/kernel/cli.ex [hunk 1]
```diff
@@ -183,54 +212,63 @@ defmodule Kernel.CLI do
...
+    "--sname",
+    "--name",
+    "--cookie",
+    "--logger-otp-reports",
+    "--logger-sasl-reports"
+  ]
+
+  defp parse_shared([erl, _ | t], config) when erl in @erl_shared_options do
+    parse_shared(t, config)
   end
```
Comment (jjcarstens) [inline lib/elixir/lib/kernel/cli.ex]: Formatter pulled this into a very large `when erl in [....]` block with lots of new lines, so I pulled it into `@erl_shared_options`.

I'm not _super_ familiar with this, but seemed like an appropriate variable name. Happy to change with better feedback

File: lib/elixir/lib/kernel/cli.ex [hunk 2]
```diff
@@ -360,38 +405,39 @@ defmodule Kernel.CLI do
...
+    wrapper(fn -> Code.eval_string(expr, []) end)
   end
 
   defp process_command({:app, app}, _config) when is_binary(app) do
     case Application.ensure_all_started(String.to_atom(app)) do
       {:error, {app, reason}} ->
-        {:error, "--app : Could not start application #{app}: " <>
-          Application.format_error(reason)}
+        msg = "--app : Could not start application #{app}: " <> Application.format_error(reason)
+        {:error, msg}
```
Comment (jjcarstens) [inline lib/elixir/lib/kernel/cli.ex]: `msg` pulled out here according to [style suggestions in this issue](https://github.com/elixir-lang/elixir/issues/6643)

File: lib/elixir/lib/kernel/cli.ex [hunk 3]
```diff
@@ -287,36 +327,40 @@ defmodule Kernel.CLI do
...
 
   defp parse_compiler(["--no-docs" | t], config) do
-    parse_compiler t, %{config | compiler_options: [{:docs, false} | config.compiler_options]}
+    parse_compiler(t, %{config | compiler_options: [{:docs, false} | config.compiler_options]})
   end
 
   defp parse_compiler(["--no-debug-info" | t], config) do
-    parse_compiler t, %{config | compiler_options: [{:debug_info, false} | config.compiler_options]}
+    updated_compiler_options = [{:debug_info, false} | config.compiler_options]
+    parse_compiler(t, %{config | compiler_options: updated_compiler_options})
```
Comment (jjcarstens) [inline lib/elixir/lib/kernel/cli.ex]: `updated_compiler_options` variable used to keep lines cleaner

File: lib/elixir/lib/kernel/cli.ex [hunk 4]
```diff
@@ -14,14 +22,17 @@ defmodule Kernel.CLI do
...
-        System.halt(1)
-      end
-    end, config.halt
+        if errors != [] do
+          Enum.each(errors, &IO.puts(:stderr, &1))
+          System.halt(1)
+        end
+      end,
+      config.halt
+    )
```
Comment (whatyouhide) [inline lib/elixir/lib/kernel/cli.ex]: Please extract the function to a separate variable and call `run(x, config.halt)`.

File: lib/elixir/lib/kernel/cli.ex (hunk 1 above)
Comment (whatyouhide) [reply lib/elixir/lib/kernel/cli.ex]: It looks good.

File: lib/elixir/lib/kernel/cli.ex [hunk 5]
```diff
@@ -55,21 +67,26 @@ defmodule Kernel.CLI do
...
     {blamed, stacktrace} = Exception.blame(kind, reason, stacktrace)
+
     iodata =
       case blamed do
         %FunctionClauseError{} ->
-          [Exception.format_banner(kind, reason, stacktrace),
-           pad(FunctionClauseError.blame(blamed, &inspect/1, &blame_match/2))]
+          [
+            Exception.format_banner(kind, reason, stacktrace),
+            pad(FunctionClauseError.blame(blamed, &inspect/1, &blame_match/2))
```
Comment (josevalim) [inline lib/elixir/lib/kernel/cli.ex]: Please extract one of those to a variable so the list fits on a single line.

File: lib/elixir/lib/kernel/cli.ex (hunk 1 above)
Comment (josevalim) [reply lib/elixir/lib/kernel/cli.ex]: I like it too, good job!

File: lib/elixir/lib/kernel/cli.ex (hunk 3 above)
Comment (josevalim) [reply lib/elixir/lib/kernel/cli.ex]: You can just call it `compiler_options`, here and below. :)

File: lib/elixir/lib/kernel/cli.ex (hunk 2 above)
Comment (josevalim) [reply lib/elixir/lib/kernel/cli.ex]: Beautiful!

**Comment by jjcarstens** [comment]
Ready for another review @whatyouhide @josevalim

**Comment by josevalim** [comment]
Merged manually because github seems to be stuck.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by josevalim** [comment]
Merged manually because GitHub was stuck!  :heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request closed without merging**